skimage/_shared/utils.py

```diff
@@ -450,3 +450,31 @@ def _fix_ndimage_mode(mode):
     if NumpyVersion(scipy.__version__) >= '1.6.0':
         mode = grid_modes.get(mode, mode)
     return mode
+
+
+def _float_type(image):
```

Comment:
From the name alone, it's very hard to know what this function will do. I suggest it should only receive the dtype, not the image, and it should be renamed to e.g. `_supported_float_type(input_dtype)`

Reply:
I was thinking the same. At the same time, one might be tempted to do `_float_type(np.float32)`, which will not work because `np.float32` is a class, not an instance. Not sure if that argument should be given weight though.

Reply:
Definitely like the suggested name. Passing the dtype instead is probably also clearer.

Regarding @FirefoxMetzger's concern we can just call `dtype = np.dtype(dtype)` within the function to make sure it is an instance.